src/main/java/build/buildfarm/instance/shard/ShardInstance.java

```diff
@@ -432,6 +451,55 @@ public void run() {
     } else {
       operationQueuer = null;
     }
+
+    if (queues.size() > 0) {
+      queueSize =
+          Gauge.build()
+              .name("queue_size")
+              .labelNames(getQueueNames(queues))
+              .help("Queue size.")
+              .register();
+    } else {
+      queueSize = null;
+    }
+
+    prometheusMetricsThread =
+        new Thread(
```

Comment:
There may be an advantage to using Java's TimerTask so we publish at a consistent rate, but its not really an issue here.

Reply:
Prefer scheduled executor service to TimerTask. Unless you believe that they mesh well.

Reply:
We should probably stay consistent and utilize the same service for all whatever one we would pick.